### DISCUSSION TITLE
Bundler/Gemfile: Gem version declaration presence cop

### DISCUSSION THREAD

**Discussion by timlkelly** [rubocop/rubocop#9720] [Ideas]
Hello, I've recently created a custom cop for a private project which I think is applicable to most codebases. I want to gauge if there's interest to include this in Rubocop. If so, I'd love to open up a pull to add it!

This cop checks that gems listed in the Gemfile include a version declaration. The reason is to prevent any unintended side effects from bundling and updating gems to new versions that weren't intended to change. I also think it's best practice to specify versions of dependencies. Here are the comments I wrote to give a little more context:

```ruby
# This cop checks that Gems specify a Version number.
# @example
#  # bad
#  gem 'rubocop'
#
#  # good
#  gem 'rubocop', '~> 1.12'
#
#  # good
#  gem 'rubocop', '>= 1.10.0'
#
#  # good
#  gem 'rubocop', '>= 1.5.0', '< 1.10.0'
#
```

**Comment by koic** [comment]
JFYI, this proposal is discussed in https://github.com/rubocop/rubocop/issues/7669.

**Comment by timlkelly** [comment]
Ah, thank you. I missed that while searching for existing issues. I'll post there.

**Comment by timlkelly** [comment]
Pull request link for discoverability: https://github.com/rubocop/rubocop/pull/9727
